src/AppInstallerCLICore/Commands/UninstallCommand.cpp

```diff
@@ -90,13 +94,19 @@ namespace AppInstaller::CLI
              execArgs.Contains(Execution::Args::Type::Id) ||
              execArgs.Contains(Execution::Args::Type::Name) ||
              execArgs.Contains(Execution::Args::Type::Moniker) ||
+             execArgs.Contains(Execution::Args::Type::ProductCode) ||
              execArgs.Contains(Execution::Args::Type::Version) ||
              execArgs.Contains(Execution::Args::Type::Channel) ||
              execArgs.Contains(Execution::Args::Type::Source) ||
              execArgs.Contains(Execution::Args::Type::Exact)))
         {
             throw CommandException(Resource::String::BothManifestAndSearchQueryProvided, "");
         }
+
+        if (execArgs.Contains(Execution::Args::Type::Purge) && execArgs.Contains(Execution::Args::Type::Preserve))
+        {
+            throw CommandException(Resource::String::BothPurgeAndPreserveFlagsProvided, "");
+        }
```

Comment:
This should probably be moved up a level into `Command::ValidateArguments` since it will apply to multiple commands, similar to how the Silent/Interactive flags are handled

Reply:
I'm going to leave this here for now as I don't want to add this validation prematurely if its only currently being used in the uninstall command. (Upgrade command only supports --purge)